src/domain/individualHarbor/forms/berthForm/BerthForm.tsx

```diff
@@ -0,0 +1,210 @@
+    length: Yup.number()
+      .typeError(t('forms.common.errors.numberType'))
+      .required(t('forms.common.errors.required')),
+    depth: Yup.mixed<number | null>().typeError(
+      t('forms.common.errors.numberType')
+    ),
+    mooringType: Yup.string().required(t('forms.common.errors.required')),
+  });
+
+  const initialValues: Berth = getBerth(data);
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={berthSchema}
+      onSubmit={values => {
+        const { width, length, mooringType, comment, isActive } = values;
+        updateBerth({
+          variables: {
+            input: {
+              id: berthId,
+              width,
+              length,
+              mooringType,
+              comment,
+              isActive,
+            },
+          },
+        }).then(() => onUpdate && onUpdate(values));
+      }}
+    >
+      {({ values, errors, handleChange, handleSubmit, setFieldValue }) => (
+        <form onSubmit={handleSubmit} className={styles.form}>
+          <Grid colsCount={3} className={styles.grid}>
+            <TextInput
+              id="pier"
+              readOnly
+              value={values.pier}
+              labelText={t('forms.berth.pier')}
+              invalid={!!errors.pier}
+              invalidText={errors.pier}
+            />
+            <TextInput
+              id="number"
+              type="number"
+              readOnly
+              value={String(values.number)}
+              labelText={t('forms.berth.number')}
+              invalid={!!errors.number}
+              invalidText={errors.number}
+            />
+            <div />
+
+            <TextInput
+              id="width"
+              type="number"
+              onChange={handleChange}
+              value={String(values.width)}
+              labelText={t('forms.berth.width')}
+              invalid={!!errors.width}
+              invalidText={errors.width}
+            />
+            <TextInput
+              id="length"
+              type="number"
+              onChange={handleChange}
+              value={String(values.length)}
+              labelText={t('forms.berth.length')}
+              invalid={!!errors.length}
+              invalidText={errors.length}
+            />
+            <TextInput
+              id="depth"
+              type="number"
+              onChange={handleChange}
+              value={String(values.depth)}
+              labelText={t('forms.berth.depth')}
+              invalid={!!errors.depth}
+              invalidText={errors.depth}
+              readOnly
+            />
+          </Grid>
+
+          <hr />
+
+          <Select
+            id="mooringType"
+            value={values.mooringType}
+            labelText={t('forms.berth.mooringType')}
+            options={Object.keys(BerthMooringType).map(mooringType => {
+              return {
+                label: t(`common.mooringTypes.${mooringType}`),
```

Comment:
Tiny detail:
```suggestion
                label: t([`common.mooringTypes.${mooringType}`, mooringType]),
```
This might be a slightly better user experience if `mooringTypes` are changed in the backend.

Reply:
Done.